lib/CArtHandler.cpp

```diff
@@ -227,14 +223,14 @@ void CGrowingArtifact::levelUpArtifact (CArtifactInstance * art)
 	b->duration = Bonus::COMMANDER_KILLED;
 	art->accumulateBonus(b);
 
-	for (auto bonus : bonusesPerLevel)
+	for(const auto & bonus : bonusesPerLevel)

```

Comment:
is bonusesPerLevel a collection of shared_ptr<Bonus> or just Bonus?

Reply:
@nullkiller does not matter I think. Const reference removes a calling a copy constructor anyway, yes, it shared_ptr case it is less optimization, but shared_ptr also alloc metadata on each copy, which is costly.